### DISCUSSION TITLE
Using Hindi for response in cloze plugin

### DISCUSSION THREAD

**Discussion by Divyanshi09** [jspsych/jsPsych#1371] [Q&A]
Hello! I have designed a Cued Recall Inhibition task to study inhibition involved in bilingual language processing. The task has to be in English as well as Hindi. One of the stages in this task involves taking responses from the participant by giving them a hint. I have used the cloze plugin for that and it is working fine. However, I cannot get it to work in Hindi. Is it possible to allow the participants to give their responses in Hindi by giving them a Hindi keyboard option or something that pops up for every trial involving that plugin (this phase is not concerned with RT)?  Also, I am using my jsPsych scripts on Gorilla, so would Gorilla play any role in that? Thanks!

**Comment by jodeleeuw** [comment]
Hi @Divyanshi09,

I don't have experience trying to get this to work in a language other than English. Can you tell us more about what problem(s) happen when you use Hindi?

**Reply by Divyanshi09** [reply]
@jodeleeuw Hi! So, the task will be in 3 steps. In the first step, I will show the participants a set of Hindi words on the screen. In the second stage, I will give them the first letter of some of the Hindi words that were shown in stage 1 and the participants are required to complete the word **in Hindi**. This is to make them memorise certain words more concretely than others. This is followed by stage 3 (which is working fine and I won't mention it again). The stimuli in Hindi is being displayed perfectly. The problem is in stage 2 where I need the participants to complete the Hindi word using Hindi script. I am using the cloze plugin. When I try filling the rest of the word, by default English letters are being used. I am wondering if there is any way by which the participants will be able to write in Hindi script instead of English (they can copy paste from Google, but that is inconvenient).
Alternatively, I also thought of replacing the cloze plugin with a canvas maybe (if there is a plugin which allows that), where the participants are shown the first letter of Hindi and they can sort of draw the rest of the word on the canvas using a pen (like the annotate function on Zoom).
I appreciate any help or ideas that can make this work!

**Reply by jodeleeuw** [reply]
After some searching around on Stack Overflow it looks like it's not possible to set the keyboard language default via JavaScript. So I imagine that participants would need to set their keyboard language to Hindi before filling out the fields. Is that reasonable? I'm sorry to be somewhat ignorant of how multilingual setups work; thanks for helping me to understand better!

**Reply by Divyanshi09** [reply]
That does sound reasonable and thanks a lot for looking it up. I am sorry for bothering you again, but can you think of any other alternative which is more convenient to make this work? I am wondering if there is a plugin that allows people to write on a canvas in response to a text stimuli using their cursor. Again, thanks for your time!
